src/openvic-simulation/economy/trading/MarketInstance.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "openvic-simulation/economy/GoodDefinition.hpp"
+#include "openvic-simulation/economy/GoodInstance.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+
+namespace OpenVic {
+	struct SellResult;
+
+	struct MarketInstance { 
+		private:
+			GoodInstanceManager const* PROPERTY(good_instance_manager);
+		public:
+			bool setup(GoodInstanceManager const& new_good_instance_manager);
+			SellResult sell(GoodDefinition const& good, const fixed_point_t quantity);
```

Comment:
Mutable as we will have to updated the GoodInstance price, demand, supply, etc at some point.